src/pynxtools/nomad/schema.py

```diff
@@ -137,6 +138,82 @@ class NexusActivityStep(ActivityStep):
     )
 
 
+class NexusIdentifier(EntityReference):
+    name = Quantity(
+        type=str,
+        description="The name of the nexus identifier field.",
+    )
+
+    def normalize(self, archive, logger):
+        from nomad.processing import Entry
+
+        def create_Entity(lab_id, archive, f_name, qunt_name):
+            entitySec = Entity()
+            entitySec.lab_id = lab_id
+            entitySec.name = qunt_name
+            entity = EntryArchive(
+                data=entitySec,
+                m_context=archive.m_context,
+                metadata=EntryMetadata(
+                    entry_type="identifier", domain="nexus", readonly=True
+                ),  # upload_id=archive.m_context.upload_id,
+            )
+            with archive.m_context.raw_file(f_name, "w") as f_obj:
+                json.dump(entity.m_to_dict(with_meta=True), f_obj)
+            archive.m_context.process_updated_raw_file(f_name)
+
+        def get_entry_reference(archive, f_name):
+            """Returns a reference to data from entry."""
+            from nomad.utils import hash
+
+            upload_id = archive.metadata.upload_id
+            entry_id = hash(upload_id, f_name)
+            return f"../uploads/{upload_id}/archive/{entry_id}#data"
+            # return f"/entries/{entry_id}/archive#/data"
+
+        f_name = f"{self.name}_{self.lab_id}.archive.json"
+        # Chekc if the entry already exists
+        entry = Entry.objects(
+            upload_id=archive.metadata.upload_id, mainfile=f_name
+        ).first()
+        if not entry:
+            create_Entity(self.lab_id, archive, f_name, self.name)
+        self.reference = get_entry_reference(archive, f_name)
+        super().normalize(archive, logger)
+
+
+class NexusReferences(ArchiveSection):
+    NXidentifiers = SubSection(
+        section_def=NexusIdentifier,
+        repeats=True,
+        description="These are the NOMAD references correspond to NeXus identifierNAME fields.",
+    )
+
+    def normalize(self, archive, logger):
+        # Consider multiple identifiers exists in the same group/section
+        identifiers = [
+            key
+            for key in self.__dict__.keys()
+            if key.startswith("identifier") and key.endswith("__field")
+        ]
+        if not identifiers:
+            return
+        self.NXidentifiers = []
+        for identifier in identifiers:
+            if not (val := getattr(self, identifier)):
+                continue
+            lab_id = (
+                re.split("([0-9a-zA-Z.]+)", val)[1]
+                + hashlib.md5(val.encode()).hexdigest()
+            )
+            identifier_path = f"{self.m_def.name}_{identifier.split('__field')[0]}"
+            logger.info(f"{lab_id} to be created")
+            logger.info(f"Identifier {val} refers to {lab_id}.")
+            nx_id = NexusIdentifier(lab_id=lab_id, name=identifier_path)
+            nx_id.normalize(archive, logger)
```

Comment:
will be called automatically, nothing to do here.

Reply:
I also assumed nomad will automatically run nomalizer for NexusIdentifier, But seems, that is not the case.